AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift

```diff
@@ -0,0 +1,432 @@
+                XCTAssertEqual(capturedPosts[index].title, queriedPost?.title)
+                XCTAssertEqual(capturedPosts[index].content, queriedPost?.content)
+                await localQueryForPosts.fulfill()
+            }
+        }
+        
+        await waitForExpectations([localQueryForPosts], timeout: networkTimeout)
+    }
+    
+    /// Perform concurrent saves and observe the data successfuly synced from cloud
+    ///
+    /// - Given: DataStore is in ready state
+    /// - When:
+    ///    - Concurrently perform Datastore.save() from 50 tasks and Datastore.query(model:where:) from 50 tasks
+    /// - Then:
+    ///    - Ensure the expected mutation event with version 1 (synced from cloud) is received
+    ///    - Query for the item is successful
+    ///
+    func testMultipleQueryByPredicate() async throws {
+        await setUp(withModels: TestModelRegistration(), logLevel: .verbose)
+        try await startAmplifyAndWaitForSync()
+
+        var posts = [Post]()
+        for index in 0 ..< concurrencyLimit {
+            let post = Post(title: "title \(index)",
+                            content: "content",
+                            createdAt: .now())
+            posts.append(post)
+        }
+        let postsSyncedToCloud = asyncExpectation(description: "All posts saved and synced to cloud",
+                                                  expectedFulfillmentCount: concurrencyLimit)
+
+        let postsCopy = posts
+        Task {
+            var postsSyncedToCloudCount = 0
+            let mutationEvents = Amplify.DataStore.observe(Post.self)
+            do {
+                for try await mutationEvent in mutationEvents {
+                    guard postsCopy.contains(where: { $0.id == mutationEvent.modelId }) else {
+                        return
+                    }
+
+                    if mutationEvent.mutationType == MutationEvent.MutationType.create.rawValue,
+                       mutationEvent.version == 1 {
+                        postsSyncedToCloudCount += 1
+                        await postsSyncedToCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
+            }
+        }
+
+        let capturedPosts = posts
+
+        DispatchQueue.concurrentPerform(iterations: concurrencyLimit) { index in
+            Task {
+                _ = try await Amplify.DataStore.save(capturedPosts[index])
+            }
+        }
+        
+        await waitForExpectations([postsSyncedToCloud], timeout: networkTimeout)
+        
+        let localQueryForPosts = asyncExpectation(description: "Query for the post is successful")
+        
+        // query by predicate where title of the post is "title 25", should be successful for
+        // only one query
+        let predicate = Post.keys.id.eq(capturedPosts[25].id).and(Post.keys.title.eq(capturedPosts[25].title))
```

Comment:
why `25` here? shouldn't the predicate change for each `index` into the `capturedPosts` array